include/COLA.hh

```diff
@@ -253,15 +254,14 @@ namespace cola {
     };
 
     inline void cola::ColaRunManager::run(int n) {
-        EventData event;
         for(int k = 0; k < n; k++) {
-            event = (*(filterAnsamble.generator))();
+            auto event = (*(filterAnsamble.generator))();
             std::queue <std::shared_ptr<VConverter>> convQ = filterAnsamble.converters;
```

Comment:
All converters are shared, but does we have any usecases for using either shared or unique pointers?

Reply:
Converters are shared because you decided to have queue copied for each pipeline invocation and unique_ptrs can't be copied (only moved). This way you have a "master queue" of converters that is left unchanged and for each invocation you duplicate it and then go through the duplicate. 

With EventData, however, the ownership of the DTO is passed through all of filters (it is generated by generator, then passed to a converter and so on, and so on, until the writer). It clearly reflects the process of filters map-reducing the data from a generator.

It is possible to move to unique_ptrs in filterAnsamble, though it would require iterating through the master queue without changing it, therefore a switch to deque is necessary. It can potentially save us some space and/or time, and the ownership principles of filters being owned only by a single object instead of being constantly associated and de-associated with temporary objects are more straitforward.

We should probably change it too.